src/Query.tsx

```diff
@@ -201,9 +215,14 @@ class Query<
   render() {
     const { children } = this.props;
     const queryResult = this.getQueryResult();
-    return children(queryResult);
+    return children(this.shouldSkip() ? undefined : queryResult);
```

Comment:
This is noted below, but we shouldn't return `undefined` here, but rather return everything we can (including `client`) so that delayed actions can be done. It also lowers the risk of error for people checking loading / data / states on an undefined vs {}